src/Game/LighthouseLight.cpp

```diff
@@ -51,6 +51,11 @@ void LighthouseLight::update()
     auto const light_locked = spotlight.lock();
 
     float const light_beam_length = glm::length(entity->transform->get_position() - light_locked->entity->transform->get_position());
+    // https://stackoverflow.com/questions/570669/checking-if-a-double-or-float-is-nan-in-c
+    if (light_beam_length != light_beam_length || light_beam_length == 0.0f)
+    {
+        return;
+    }
```

Comment:
When exactly does this happen? I presume when `entity->transform->get_position() == light_locked->entity->transform->get_position()`? If yes, then we can just check for this situation instead of checking for `NAN`

Reply:
I investigated a little bit more and it seems like we had a bug in the Input system. Let's fix the issue in its core.